tinygrad/tensor.py

```diff
@@ -441,6 +443,7 @@ def full(shape:Tuple[sint, ...], fill_value:ConstType, **kwargs):
     print(Tensor.full((2, 3), False).numpy())
     ```
     """
+    if not all(s >= 0 for s in argfix(shape)): raise RuntimeError(f"Trying to create view with negative dimension: {shape=}")
```

Comment:
argfix is called twice, also don't just copy paste error messages. the message does not make sense in Tensor

Reply:
Also, python won't let me use walrus operator in that context:

```
if not all(s >= 0 for s in (new_shape := argfix(shape))): raise RuntimeError(f"Trying to create Tensor with negative dimension: {shape=}")
```
highlights `:=` as having wrong